File: src/main/java/com/example/budget/domain/budget/entity/Budget.java
package com.example.budget.domain.budget.entity;

import com.example.budget.domain.user.entity.User;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "budgets")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Budget {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private Long amount;

    @Column(name = "budget_year", nullable = false)
    private Integer year;

    @Column(name = "budget_month", nullable = false)
    private Integer month;

    @Builder
    public Budget(User user, Long amount, Integer year, Integer month) {
        this.user = user;
        this.amount = amount;
        this.year = year;
        this.month = month;
    }

    public void updateAmount(Long amount) {
        this.amount = amount;
    }
}

File: gagaibu_flutter/lib/models/budget.dart
class Budget {
  final int? id;
  final int userId;
  final int amount;
  final int year;
  final int month;

  Budget({
    this.id,
    required this.userId,
    required this.amount,
    required this.year,
    required this.month,
  });

  factory Budget.fromJson(Map<String, dynamic> json) {
    return Budget(
      id: json['id'] as int?,
      userId: json['userId'] as int,
      amount: json['amount'] as int,
      year: json['year'] as int,
      month: json['month'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'userId': userId,
      'amount': amount,
      'year': year,
      'month': month,
    };
  }
}
